cron/createFDQNBlockList.py

```diff
@@ -0,0 +1,248 @@
+#         if not ipcidr.ip.is_loopback() and not ipcidr.ip.is_private() and not ipcidr.ip.is_reserved():
+#             for whitelist_range in options.ipwhitelist:
+#                 whitelist_network = netaddr.IPNetwork(whitelist_range)
+#                 if ipcidr in whitelist_network:
+#                     logger.debug(str(ipcidr) + " is whitelisted as part of " + str(whitelist_network))
+#                     whitelisted = True
+
+#             #strip any host bits 192.168.10/24 -> 192.168.0/24
+#             ipcidrnet=str(ipcidr.cidr)
+#             if ipcidrnet not in iplist and not whitelisted:
+#                 iplist.append(ipcidrnet)
+#         else:
+#             logger.debug('invalid:' + ip)
+#     return iplist
+
+def parse_fqdn_whitelist(fqdn_whitelist_location):
+    fqdns = []
+    with open(fqdn_whitelist_location, "r") as text_file:
+        for line in text_file:
+            line=line.strip().strip("'").strip('"')
+            if isFQDN(line):
+                fqdns.append(line)
+    return fqdns
+
+def main():
+    logger.debug('starting')
+    logger.debug(options)
+    try:
+        # connect to mongo
+        client = MongoClient(options.mongohost, options.mongoport)
+        mozdefdb = client.meteor
+        fqdnblocklist = mozdefdb['fqdnblocklist']
+        attackers=mozdefdb['attackers']
+        # ensure indexes
+        fqdnblocklist.create_index([('dateExpiring',-1)])
+
+        # TODO: determine if this is relevant in this context
+        # attackers.create_index([('lastseentimestamp',-1)])
+        # attackers.create_index([('category',1)])
+
+        # First, gather IP addresses from recent attackers and add to the block list
+        # attackerIPList = aggregateAttackerIPs(attackers)
+
+        # add attacker IPs to the blocklist
+        # first delete ones we've created from an attacker
+        # ipblocklist.delete_many({'creator': 'mozdef','reference':'attacker'})
+
+        # delete any that expired
+        fqdnblocklist.delete_many({'dateExpiring': {"$lte": datetime.utcnow()-timedelta(days=options.expireage)}})
+
+        # add the aggregations we've found recently
+        # for ip in attackerIPList:
+        #     ipblocklist.insert_one(
+        #         {'_id': genMeteorID(),
+        #          'address':ip,
+        #          'reference': 'attacker',
+        #          'creator':'mozdef',
+        #          'dateAdded': datetime.utcnow()})
+
+        # Lastly, export the combined blocklist
+        fqdnCursor=mozdefdb['fqdnblocklist'].aggregate([
+                {"$sort": {"dateAdded": -1}},
+                {"$match": {"address": {"$exists": True}}},
+                {"$match":
+                    {"$or":[
+                        {"dateExpiring": {"$gte": datetime.utcnow()}},
+                        {"dateExpiring": {"$exists": False}},
+                    ]},
+                },
+                {"$project":{"address":1}},
+                {"$limit": options.iplimit}
```

Comment:
probably needs a new option. The option is just to limit any list to X entries.

Reply:
changed to fqdnlimit, but not sure if I need to worry about namespacing $limit